packages/terra-table/src/subcomponents/_HeaderCell.jsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './HeaderCell.module.scss';
+import {
+  styleFromWidth,
+  wrappedOnClickForItem,
+  wrappedOnKeyDownForItem,
+  wrappedEventCallback,
+} from './utils';
+import widthShape from '../proptypes/widthShape';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Content to be displayed for the column header.
+   */
+  children: PropTypes.node,
+  /**
+   * The associated metaData to be provided in the onSelect callback.
+   */
+  // eslint-disable-next-line react/forbid-prop-types
+  metaData: PropTypes.object,
+  /**
+   * Function callback returning the html node for the header cell.
+   */
+  refCallback: PropTypes.func,
+  /**
+   * Whether or not the cell's inner containing element responsible for handling table's default padding is removed.
+   * This is useful to optimize the DOM for either a table without padding or to optimize a cell whose custom content is providing its own padding.
+   */
+  removeInner: PropTypes.bool,
+  /**
+   * Whether or not the sort direction is descending. False indicates ascending.
+   */
+  isSortDesc: PropTypes.bool,
+  /**
+   * Whether or not the header cell should display as an actively sorted cell.
+   */
+  isSortActive: PropTypes.bool,
+  /**
+   * Function callback associated to a pure cell click/action, potentially for selection, etc.
```

Comment:
is the event and/or metaData passed back when executed?

Reply:
Updated.